hearth/lib/hearth/checksums.rb

```diff
@@ -11,20 +19,38 @@ module Checksums
     # @param [File, Tempfile, StringIO#read, String] value
     # @return [String<MD5>]
     def self.md5(value)
-      if value.is_a?(File) || value.is_a?(Tempfile)
-        OpenSSL::Digest.new('MD5').file(value).base64digest
-      elsif value.respond_to?(:read)
-        md5 = OpenSSL::Digest.new('MD5')
+      digest = algorithm_for('MD5')
+      if value.respond_to?(:read)
         loop do
           chunk = value.read(CHUNK_SIZE)
           break unless chunk
 
-          md5.update(chunk)
+          digest.update(chunk)
         end
         value.rewind
-        md5.base64digest
+        digest.base64digest
       else
-        OpenSSL::Digest.new('MD5').base64digest(value)
+        digest.tap { |d| d.update(value) }.base64digest
```

Comment:
Can we have a base64digest method on digest like before?

Reply:
You mean allow it to take a value and both update and return it?

We technically do, because it just aliases the underlying method.  I was using the SRA defined interface here, but we could just do `d.base64digest(value)`

Reply:
Yes I would prefer that